searchlib/src/main/java/com/yahoo/searchlib/gbdt/XmlHelper.java

```diff
@@ -27,38 +27,25 @@ abstract class XmlHelper {
     private static final Charset UTF8 = Charset.forName("UTF-8");
 
     public static Element parseXml(String xml)
-            throws ParserConfigurationException, IOException, SAXException
+            throws IOException, SAXException
     {
         return parseXmlStream(new ByteArrayInputStream(xml.getBytes(UTF8)));
     }
 
     public static Element parseXmlFile(String fileName)
-            throws ParserConfigurationException, IOException, SAXException
+            throws IOException, SAXException
     {
         return parseXmlStream(new FileInputStream(fileName));
     }
 
     public static Element parseXmlStream(InputStream in)
-            throws ParserConfigurationException, IOException, SAXException
+            throws IOException, SAXException
     {
-        DocumentBuilderFactory factory = createDocumentBuilderFactory();
-        DocumentBuilder builder = factory.newDocumentBuilder();
+        DocumentBuilder builder = XML.getDocumentBuilder();
         Document doc = builder.parse(in);
         return doc.getDocumentElement();
     }
 
-    private static DocumentBuilderFactory createDocumentBuilderFactory() throws ParserConfigurationException {
-        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-        factory.setNamespaceAware(true);
-        factory.setXIncludeAware(false);
-
-        // XXE prevention
-        factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
-        factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
-        factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
```

Comment:
Seems like this feature is not set in XML.getDocumentBuilder methods. Intended?

Reply:
This PR is a bit more strict, need to verify that we can indeed do this. I'll split this in 2 PRs, and we can merge the first part now.